modules/e2ee/Context.js

```diff
@@ -227,16 +246,43 @@ export class Context {
                     new Uint8Array(newData)).then(async authTag => {
                     const truncatedAuthTag = new Uint8Array(authTag, 0, DIGEST_LENGTH[encodedFrame.type]);
 
-
                     // Set the truncated authentication tag.
                     newUint8.set(truncatedAuthTag, UNENCRYPTED_BYTES[encodedFrame.type] + cipherText.byteLength);
 
                     // Sign with the long-term signature key.
                     if (signatureLength) {
+                        const numberOfOldTags = this._frameSignatures.has(ssrc)
+                            ? this._frameSignatures.get(ssrc).length
+                            : 0;
+                        const signatureData = new Uint8Array(
+                            (numberOfOldTags * DIGEST_LENGTH[encodedFrame.type]) + truncatedAuthTag.byteLength);
+
+                        signatureData.set(truncatedAuthTag, 0);
+                        let offset = truncatedAuthTag.byteLength;
+
+                        for (const oldAuthTag of this._frameSignatures.get(ssrc) || []) {
+                            signatureData.set(oldAuthTag, offset);
+                            offset += oldAuthTag.byteLength;
+                        }
+
+                        this._frameSignatures.set(ssrc, []);
+
                         const signature = await crypto.subtle.sign(this._signatureOptions,
-                            this._signatureKey, truncatedAuthTag);
+                            this._signatureKey, signatureData);
 
                         newUint8.set(new Uint8Array(signature), newUint8.byteLength - signature.byteLength - 1);
+
+                        // This count excludes the new authentication tag (which is always there)
+                        newUint8[newUint8.byteLength - 1 - this._signatureOptions.byteLength
```

Comment:
i *think* we can just define this including the authentication tag and save some "+ 1" on the receiver.